python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -61,13 +61,20 @@ def __init__(self, config, default=None):
             else:
                 self.templates[name] = FileTemplate(templateStr)
 
-    def getTemplate(self, datasetTypeName):
+    def getTemplate(self, entity):
         """Retrieve the `FileTemplate` associated with the dataset type.
 
+        If the lookup name corresponds to a component the base name for
+        the component will be examined if the full component name does
+        not match.
+
         Parameters
         ----------
-        datasetTypeName : `str`
-            Dataset type name.
+        entity : `DatasetType`, `DatasetRef`, or `StorageClass`
+            Instance to use to look for a corresponding template.
+            A `DatasetType` name or a `StorageClass` name will be used
+            depending on the supplied entity. Priority is given to a
+            `DatasetType` name.
```

Comment:
Just a thought for the future: template lookup by StorageClass is probably less useful in practice than template lookup by "set of DataUnits".  In other words, I probably want my templates for Tract+Patch Datasets to be more similar than my templates for all SourceCatalog Datasets.  But the existing support for optional replacement tokens in templates will hopefully make that a non-issue.